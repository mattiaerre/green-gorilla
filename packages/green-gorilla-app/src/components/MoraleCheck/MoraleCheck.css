ul.morale-check {
  list-style: none;
  padding: 1em 5em;
  margin: 0;
}

.morale-check > li {
  padding: 1em;
  display: flex;
  align-items: center;
  justify-content: center;
}

span.emoji {
  font-size: 2em;
  margin: 0 0.5em;
}
