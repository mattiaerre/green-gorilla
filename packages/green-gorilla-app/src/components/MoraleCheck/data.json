[
  {
    "id": "awesome",
    "description": "I'm feeling awesome",
    "emoji": "😎"
  },
  {
    "id": "good",
    "description": "I'm feeling good",
    "emoji": "👍"
  },
  {
    "id": "not-great",
    "description": "I'm not feeling great",
    "emoji": "😕"
  },
  {
    "id": "unhappy",
    "description": "I'm feeling unhappy",
    "emoji": "😩"
  }
]
