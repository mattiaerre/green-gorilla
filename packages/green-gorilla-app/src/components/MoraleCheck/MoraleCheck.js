import React from 'react';
import data from './data';
import './MoraleCheck.css';

const MoraleCheck = ({ onClick }) => {
  const items = data.map(({ id, description, emoji }) => (
    <li key={id} className={id} onClick={() => { onClick(id) }}>
      <span className="emoji">{emoji}</span>
      <span className="description">{description}</span>
    </li>)
  );

  return (
    <ul className="morale-check">
      {items}
    </ul>
  );
};

export default MoraleCheck;
