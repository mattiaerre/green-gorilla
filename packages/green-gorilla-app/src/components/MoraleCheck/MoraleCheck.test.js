import React from 'react';
import { create } from 'react-test-renderer';
import MoraleCheck from './MoraleCheck';

describe('<MoraleCheck />', () => {
  test('should match snapshot', () => {
    const component = create(
      <MoraleCheck handleOnCLick={jest.fn()} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
