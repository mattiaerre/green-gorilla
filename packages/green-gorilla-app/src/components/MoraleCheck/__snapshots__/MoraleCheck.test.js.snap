// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MoraleCheck /> should match snapshot 1`] = `
<ul
  className="morale-check"
>
  <li
    className="awesome"
    onClick={[Function]}
  >
    <span
      className="emoji"
    >
      😎
    </span>
    <span
      className="description"
    >
      I'm feeling awesome
    </span>
  </li>
  <li
    className="good"
    onClick={[Function]}
  >
    <span
      className="emoji"
    >
      👍
    </span>
    <span
      className="description"
    >
      I'm feeling good
    </span>
  </li>
  <li
    className="not-great"
    onClick={[Function]}
  >
    <span
      className="emoji"
    >
      😕
    </span>
    <span
      className="description"
    >
      I'm not feeling great
    </span>
  </li>
  <li
    className="unhappy"
    onClick={[Function]}
  >
    <span
      className="emoji"
    >
      😩
    </span>
    <span
      className="description"
    >
      I'm feeling unhappy
    </span>
  </li>
</ul>
`;
